### PR TITLE
planner: fix the wrong schema.Clone implementation

### LINKED ISSUE #57635: unexpected assert failed in TestPlanCacheClone
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

Run TestPlanCacheClone using https://github.com/pingcap/tidb/pull/57597/commits/c8fee91d22c9ff2b4e9fec8989d0f2c933ee73c6 which change non-partition table to partition table.

### 2. What did you expect to see? (Required)
Test passes.
### 3. What did you see instead (Required)
img_v3_02gr_262c8e93-fa20-4cb7-a4c4-27486b1f533g

### 4. What is your TiDB version? (Required)
master

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: close #57635

Problem Summary: planner: fix the wrong schema.Clone implementation

### What changed and how does it work?

planner: fix the wrong schema.Clone implementation

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/expression/schema.go [hunk 1]
```diff
@@ -62,7 +62,9 @@ func (s *Schema) String() string {
 	for _, key := range s.Keys {
 		ukStrs = append(ukStrs, key.String())
 	}
-	return "Column: [" + strings.Join(colStrs, ",") + "] Unique key: [" + strings.Join(ukStrs, ",") + "]"
+	return "Column: [" + strings.Join(colStrs, ",") +
+		"] Key: [" + strings.Join(ukStrs, ",") +
+		"] Unique key: [" + strings.Join(ukStrs, ",") + "]"
```
Comment (windtalker) [inline pkg/expression/schema.go:67]: why use the same `ukStrs` for `Key` and `Unique key`?

File: pkg/expression/schema_test.go [hunk 2]
```diff
@@ -48,12 +48,29 @@ func (s *schemaGenerator) generateSchema(colCount int) *Schema {
...
+
+	uniKeys := make([]KeyInfo, 0, len(schema.Columns)-1)
+	for i := 0; i < len(schema.Columns)-1; i++ {
+		uniKeys = append(uniKeys, []*Column{schema.Columns[i]})
+	}
+	schema.SetUniqueKeys(uniKeys)
+
+	clonedSchema := schema.Clone()
+	require.Equal(t, schema.String(), clonedSchema.String())
+	require.True(t, fmt.Sprintf("%p", schema.Keys) != fmt.Sprintf("%p", clonedSchema.Keys))
```
Comment (lance6716) [inline pkg/expression/schema_test.go:64]: can use `require.NotSame`

**Review by windtalker — APPROVED** [review]
lgtm

File: pkg/expression/schema_test.go (hunk 2 above)
Comment (qw4990) [reply pkg/expression/schema_test.go]: done

File: pkg/expression/schema.go [hunk 3]
```diff
@@ -79,7 +79,14 @@ func (s *Schema) Clone() *Schema {
 		keys = append(keys, key.Clone())
 	}
 	schema := NewSchema(cols...)
-	schema.SetUniqueKeys(keys)
```
[the remark this answers is not in the document]
Comment (qw4990) [reply pkg/expression/schema.go:82]: done

**Review by lance6716 — APPROVED** [review]

**Pull request merged**